core/client/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java

```diff
@@ -66,15 +66,17 @@ public void readFromLocal() throws Exception {
           .thenReturn(InetSocketAddress.createUnresolved(clientHostname, 0));
       BlockInStream stream = BlockInStream.createUfsBlockInStream(context, "ufsPath", blockId, 100,
           0, 0, new WorkerNetAddress(), InStreamOptions.defaults());
-      // The client hostname matches the worker hostname, so the stream should go to a local file.
-      Assert.assertTrue(stream.isShortCircuit());
+      // The client hostname matches the worker hostname, but the stream goes to an Alluxio worker
+      // because it initially tried to read from UFS.
+      Assert.assertFalse(stream.isShortCircuit());
```

Comment:
This is related to your comment made in BlockInStream.java where I removed the optimization to read from local when trying to read from ufs.